compaction/compaction_manager.cc

```diff
@@ -334,7 +334,7 @@ class compaction_manager::major_compaction_task : public compaction_manager::tas
         co_await coroutine::switch_to(_cm._compaction_controller.sg());
 
         switch_state(state::pending);
-        auto units = co_await get_units(_cm._maintenance_ops_sem, 1, _compaction_data.abort);
+        auto units = co_await get_units(_cm._maintenance_ops_sem);
```

Comment:
It took me a minute to understand what I'm seeing here. The old code used Seastar's global function get_units(), and here you are calling a local function with the same name that basically hide's Seastar function. I think it's confusing, and makes understanding this code unnecessarily difficult. But it's not wrong.

Reply:
correct

Reply:
Shall I commit it like this, or are you thinking of renaming this function?

Reply:
I guess I got used to c++ polymorphism so I don't find it confusing, but I don't mind renaming it if it might confuse others.